index.js

```diff
@@ -0,0 +1,115 @@
+          <h2 class="${isDarkMode ? 'dark-country-title' : 'country-title'}">${country.name}</h2>
+          <ul class="country-brief">
+            <li><strong>population: </strong>${country.population}</li>
+            <li><strong>Region: </strong>${country.region}</li>
+            <li><strong>capital: </strong>${country.capital}</li>
+          </ul>
+        </div>
+      </a>
+    `;
+        countriesContainer.innerHTML += countryDiv;
+    });
+}
+
+
+const darkMode = () => {
+    const lightBody = document.querySelector('.body');
+    const darkBody = document.querySelector('.dark-theme');
+    const headerBackgroundColor = getComputedStyle(document.querySelector('header')).backgroundColor;
+    const countryDivs = document.querySelectorAll('.country-info');
+    const countryNames = document.querySelectorAll('.country-title');
+    const darkCountryDivs = document.querySelectorAll('.dark-country-info');
+    const darkCountryNames = document.querySelectorAll('.dark-country-title');
+
+    if (headerBackgroundColor === 'rgb(255, 255, 255)') {
+        lightBody.className = 'dark-theme'
+        countryDivs.forEach(div => {
+            div.className = 'dark-country-info'
+        });
+        countryNames.forEach(div => {
+            div.className = 'dark-country-title'
+        });
+
+    } else {
+        darkBody.className = 'body';
+        darkCountryDivs.forEach(div => {
+            div.className = 'country-info'
+        });
+        darkCountryNames.forEach(div => {
+            div.className = 'country-title'
+        });
+    }
+};
+
+
+const setDetailsOfCountry = (flag, name, population, region, capital) => {
+    const countryDetails = { flag, name, population, region, capital };
+    localStorage.setItem("countryDetails", JSON.stringify(countryDetails));
+    console.log("Country details saved:", countryDetails);
+};
+
+const openDropDown = () => {
+    const dropDown = document.querySelector('.dropdown-wrapper');
+    dropDown.addEventListener('click', () => {
+        dropDown.classList.toggle('open');
+    })
+}
+openDropDown();
+
+const filterByContinants = () => {
+    const regions = document.querySelectorAll('.dropdown-body li');
+    regions.forEach((li) => {
+        const chosenRegion = li.getAttribute('data-region');
+        li.addEventListener('click', () => {
+            fetch('./CountriesData.json')
+                .then(response => response.json())
+                .then(countriesJson => {
+                    let filteredCountries = countriesJson;
+                    if (chosenRegion == 'africa') {
+                        filteredCountries = filteredCountries.filter(country => country.region === 'Africa');
+                    }
+                    if (chosenRegion == 'america') {
+                        filteredCountries = filteredCountries.filter(country => country.region === 'Americas');
+                    }
+                    if (chosenRegion == 'asia') {
```

Comment:
This code repeat itself, you can do something like this : 


```js
const regionMap: Record<string, string> = {
    america: 'Americas',
    asia: 'Asia'
};

if (chosenRegion in regionMap) {
    filteredCountries = filteredCountries.filter(country => country.region === regionMap[chosenRegion]);
}
```

In this way you will not repeat yourself again and again